src/file-stat.c

```diff
@@ -191,8 +200,15 @@ stat_initialize(mrb_state *mrb, mrb_value self)
     tmp = mrb_convert_type(mrb, fname, MRB_TT_STRING, "String", "to_str");
   }
   path = mrb_str_to_cstr(mrb, tmp);
-  if (STAT(path, &st) == -1) {
-    mrb_sys_fail(mrb, path);
+  {
+    char *locale_path;
+    int stat_result;
+
+    locale_path = mrb_locale_from_utf8(path, -1);
```

Comment:
Is not `mrb_locale_free(locale_path)` necessary?

Reply:
Oh, it's needed.

Reply:
I've added it.